src/server/system_services/pool_controllers.js

```diff
@@ -68,88 +67,48 @@ class ManagedStatefulSetPoolController extends PoolController {
             JSON.stringify(pool_k8s_conf, null, 2));
 
-        await kube_utils.apply_conf(pool_k8s_conf);
-        return yaml_utils.stringify(pool_k8s_conf);
+        await KubeStore.instance.create_secret(secret_k8s_conf);
+        await KubeStore.instance.create_backingstore(pool_k8s_conf);
+        const secret_yaml = yaml.stringify(secret_k8s_conf, 6, 2);
+        const backstore_yaml = yaml.stringify(pool_k8s_conf, 6, 2);
+
+        return `${secret_yaml}---\n${backstore_yaml}`;
     }
 
     async scale(host_count) {
-        const pod_count = (await this._list_pods()).length;
-        const delta = host_count - pod_count;
+        const current_host_count = await this.get_current_volume_number();
+        const delta = host_count - current_host_count;
         if (delta === 0) return;
 
         // scale the stateful set
-        const { _statefulset_name } = this;
         dbg.log0(`ManagedStatefulSetPoolController::scale: scaling ${
             delta > 0 ? 'up' : 'down'
-        } stateful set  ${
-            _statefulset_name
+        } backing store set  ${
+            this.pool_name
         } from ${
-            pod_count
+            current_host_count
         } replicas to ${
             host_count
         } replicas`);
 
         if (host_count > 0) {
-            await kube_utils.patch_resource('StatefulSet', _statefulset_name, { spec: { replicas: host_count } });
+            await KubeStore.instance.patch_backingstore(this.pool_name, { spec: { pvPool: { numVolumes: host_count } } });
         } else {
-            await kube_utils.delete_resource('StatefulSet', _statefulset_name);
-        }
-
-        if (delta < 0) {
-            // If we are scaling down we should also clean the pvc resource claimed by
-            // the terminated pods.
-            const pods = await this._list_pods();
-            await Promise.all(pods
-                .filter(pod => {
-                    const pod_name = pod.metadata.name;
-                    const pod_index = Number(pod_name.slice(pod_name.lastIndexOf('-') + 1));
-                    return pod_index >= host_count;
-                })
-                .map(async pod => {
-                    const pod_name = pod.metadata.name;
-
-                    // Wait for the pod to be deleted.
-                    dbg.log0(`ManagedStatefulSetPoolController::scale: waiting for pod ${pod_name} to be deleted`);
-                    await kube_utils.wait_for_delete('pod', pod_name, 15 * 60 * 1000);
-
-                    // Delete the pvc claimed by the pod.
-                    const pvc_name = `noobaastorage-${pod_name}`;
-                    dbg.log0(`ManagedStatefulSetPoolController::scale: deleting pvc ${pvc_name}`);
-                    return kube_utils.delete_resource('pvc', pvc_name);
-                })
-           );
+            await KubeStore.instance.delete_backingstore(this.pool_name);
```

Comment:
Again you need to check the return value in case the backing_store deletion failed and at least put something in the logs

Reply:
it will throw exception